Libs/Mesh/Mesh.cpp

```diff
@@ -967,6 +989,599 @@ std::ostream& operator<<(std::ostream &os, const Mesh& mesh)
+                // Add points in path
+                dijkstra->SetStartVertex(lastId);
+                dijkstra->SetEndVertex(ptid);
+                dijkstra->Update();
+                vtkSmartPointer<vtkIdList> idL = dijkstra->GetIdList();
+                for(size_t j = 1; j < idL->GetNumberOfIds(); j++){
+                    vtkIdType id = idL->GetId(j);
+                    double pathpt[3];
+                    mesh->GetPoint(id, pathpt);
+                    selectionPoints->InsertNextPoint(pathpt[0],pathpt[1],pathpt[2]);
+                    boundaryVerts.push_back(id);
+                }
+            }
+            lastId = ptid;
+        }
+
+        std::cout << "Number of boundary vertices " << boundaryVerts.size() << std::endl;
+
+        vtkSmartPointer<vtkSelectPolyData> select = vtkSelectPolyData::New();
+        select->SetLoop(selectionPoints);
+        select->SetInputData(this->mesh);
+        select->GenerateSelectionScalarsOn();
+        select->SetSelectionModeToLargestRegion();
+
+        // Clipping mesh
+        vtkSmartPointer<vtkClipPolyData> selectclip = vtkClipPolyData::New();
+        selectclip->SetInputConnection(select->GetOutputPort());
+        selectclip->SetValue(0.0);
+
+        selectclip->Update();
+
+        MeshType halfmesh = selectclip->GetOutput();
+
+        vtkSmartPointer<vtkDoubleArray> inout = computeInOutForFFCs(query, halfmesh);
+
+
+
+        vtkSmartPointer<vtkDoubleArray> values = vtkSmartPointer<vtkDoubleArray>::New();
+        vtkSmartPointer<vtkDoubleArray> absvalues = setDistanceToBoundaryValueFieldForFFCs(values, points, boundaryVerts, inout, V, F, dom);
+
+        this->mesh->GetPointData()->SetActiveScalars("value");
+
+        std::vector<Eigen::Matrix3d> face_grad_ = this->setGradientFieldForFFCs(absvalues, V, F);
+
+        //this->visualizeVectorFieldForFFCs(values, face_grad_, V, F);
+
+    } // Per boundary for loop end
+
+    locator = vtkSmartPointer<vtkCellLocator>::New();
+    locator->SetDataSet(this->mesh);
+    locator->BuildLocator();
+
+    // Write mesh for debug purposes
+//    std::string fnin = "dev/mesh_" + std::to_string(dom) + "_" + std::to_string(num) + "_in.vtk";
+//    this->write(fnin);
+
+     return true;
+}
+
+// Copied directly from Meshwrapper
+Eigen::Vector3d Mesh::ComputeBarycentricCoordinates(const Eigen::Vector3d& pt, int face) const
+{
+  double closest[3];
+  int sub_id;
+  double pcoords[3];
+  double dist2;
+  Eigen::Vector3d bary;
+  this->mesh->GetCell(face)->EvaluatePosition(pt.data(), closest, sub_id, pcoords, dist2, bary.data());
+  return bary;
+}
+
+double Mesh::GetFFCValue(Eigen::Vector3d query){
+    locator->BuildLocator();
+
+    double closestPoint[3];
+    vtkIdType cellId;
+    int subId;
+    double dist;
+    locator->FindClosestPoint(query.data(), closestPoint, cellId, subId, dist);
+
```

Comment:
Mesh.closestPoint can be used

Reply:
Same as above. I want my own member locator that I can reuse.